src/test/java/net/openhft/chronicle/wire/WiresTest.java

```diff
@@ -154,7 +156,76 @@ public void unknownType2() {
                 "  big: 0.128,\n" +
                 "  also: extra\n" +
                 "}\n", tv.toString());
+    }
+
+    @Test(expected = ClassNotFoundRuntimeException.class)
+    public void unknownType2WarnThrows() {
+        wiresThrowCNFRE(false);
+        Wires.GENERATE_TUPLES = false;
+
+        String text = "!FourValues {\n" +
+                "  string: Hello,\n" +
+                "  num: 123,\n" +
+                "  big: 1E6,\n" +
+                "  also: extra\n" +
+                "}\n";
+        ThreeValues tv = Marshallable.fromString(ThreeValues.class, text);
+        assertEquals(text, tv.toString());
+        assertEquals("Hello", tv.string());
+        tv.string("Hello World");
+        assertEquals("Hello World", tv.string());
+
+        assertEquals(123, tv.num());
+        tv.num(1234);
+        assertEquals(1234, tv.num());
+
+        assertEquals(1e6, tv.big(), 0.0);
+        tv.big(0.128);
+        assertEquals(0.128, tv.big(), 0.0);
+
+        assertEquals("!FourValues {\n" +
+                "  string: Hello World,\n" +
+                "  num: !int 1234,\n" +
+                "  big: 0.128,\n" +
+                "  also: extra\n" +
+                "}\n", tv.toString());
+    }
+
+    @Test(expected = ClassNotFoundRuntimeException.class)
```

Comment:
Suggest that for new code we use `assertThrows` as this gives finer grained control over where the exception thrown from. Just a suggestion - I don't want to hold things up.